ergo-wallet/src/test/scala/org/ergoplatform/wallet/boxes/DefaultBoxSelectorSpec.scala

```diff
@@ -218,4 +215,28 @@ class DefaultBoxSelectorSpec extends AnyPropSpec with Matchers with EitherValues
     s2.right.get.changeBoxes.size shouldBe 3
     s2.right.get.changeBoxes.exists(_.tokens.size == 1) shouldBe true
   }
+
+  // test which shows that https://github.com/ergoplatform/ergo/issues/1644 fixed
+  property("i1644: should collect needed inputs when needed for change in presence of assets") {
+    val tokenData = genTokens(1)
+
+    val ergValue = 10 * MinBoxValue
+
+    val box1 = testBox(ergValue, TrueLeaf, StartHeight, tokenData)
+    val uBox1 = TrackedBox(parentTx, 0, Some(100), box1, Set(PaymentsScanId))
+    val box2 = testBox(ergValue, TrueLeaf, StartHeight)
+    val uBox2 = TrackedBox(parentTx, 0, Some(100), box2, Set(PaymentsScanId))
+
+    val s1 = select(Iterator(uBox1, uBox2), noFilter, ergValue, Map.empty)
+    s1 shouldBe 'right
+    s1.right.get.changeBoxes.size shouldBe 1
```

Comment:
Suggest to also check that there is the token in the change box.

Reply:
ok, done